### ISSUE TITLE
fix: circular dependencies result in endless loop + out of memory crash

### ISSUE THREAD

**Issue by petermetz** [davideicardi/live-plugin-manager#54]
I tried installing a plugin that has [es6-iterator](https://www.npmjs.com/package/es6-iterator) as a dependency which has [es5-ext](https://www.npmjs.com/package/es5-ext) as a dependency which has [es6-iterator](https://www.npmjs.com/package/es6-iterator) and so there's a cycle in the dependency graph that I think live-plugin-manager needs a circuit breaker for. 
It might be that some additional things are needed to make sure it is handled correctly, if anyone has any information about how npm/yarn handle these cases (on top of having the circuit breaker) that would be great to know!

**Comment by petermetz** [comment]
@davideicardi 

> Circular requires should be supported, but not sure about circular dependencies.

Fair point IMO. The only reason why I'd still consider dealing with it due to wanting to be consistent with the other package managers (both npm and yarn are able to install these dependencies). Of course I also understand 100% that if being consistent in all fronts like that is not part of the vision of the project.

> AS a workaround you can probably install es6-iterator and es5-ext manually and then mark them as a static dependencies? (see staticDependencies options)

I had to go another route so I'm unable to test it anymore unfortunately. (I'm using npm's `--prefix` parameter to install plugins "somewhere else on the file-system")  
Based on the above I'm also okay with just closing this of course, if no one else had this issue and I've moved on then I don't want to take up your time!

**Comment by davideicardi** [comment]
For me we can leave the issue open for the future, so I can monitor it.

**Comment by davue** [comment]
The same issue happens with every package that has one of these dependencies at some point:
- `has-proto`
- `call-bind`
- `es-define-property`
- `get-intrinsic`

**Comment by davideicardi** [comment]
Thank you @davue for the details, this could be useful. I will try to investigate. As usual any help is always appreciated
